File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/AtmoAuraIqaApi.java
package wangdaye.com.geometricweather.weather.apis;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import wangdaye.com.geometricweather.weather.json.atmoaura.AtmoAuraPointResult;

/**
 * API Atmo AURA
 * Covers Auvergne-Rhône-Alpes
 */

public interface AtmoAuraIqaApi {

    @GET("air2go/v3/point?with_list=true")
    Observable<AtmoAuraPointResult> getPointDetails(@Query("api_token") String api_token,
                                                    @Query("x") String longitude,
                                                    @Query("y") String latitude,
                                                    @Query("datetime_echeance") String datetime_echeance);
}

File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/AtmoAuraIqaApi.kt
package wangdaye.com.geometricweather.weather.apis

import retrofit2.http.GET
import retrofit2.http.Query
import wangdaye.com.geometricweather.weather.json.atmoaura.AtmoAuraQAResult

/**
 * API Atmo AURA
 * Covers Auvergne-Rhône-Alpes
 */
interface AtmoAuraIqaApi {

    @GET("air2go/full_request")
    suspend fun getQAFull(
            @Query("api_token") api_token: String,
            @Query("latitude") latitude: String,
            @Query("longitude") longitude: String
    ): AtmoAuraQAResult
}